libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,140 @@
+import { NgControl, NgForm } from '@angular/forms';
+import { Component, ChangeDetectorRef, Self, Optional, Input } from '@angular/core';
+import { stateType, compareObjects } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput {
+    /** Sets text of control label. */
+    @Input()
+    label: string;
+
+    /** Enables controls third state. */
+    @Input()
+    tristate: boolean = false;
+
+    /** Allows to prevent user from manually selecting controls third state. */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /**
+     * checkbox values, Sets values returned by control
+     */
+    @Input('values')
+    set checkboxvalues(checkboxValues: FdCheckboxValues) {
+        this.values = { ...this.values, ...checkboxValues };
+        this.checkboxValue = this.checkboxValue ? this.checkboxValue : this.values.falseValue;
+        this._setState();
+        this.onChange(this.checkboxValue);
+    }
+
+    /** */
+    @Input()
+    set value(value: any) {
+        if (value) {
+            this.setValue(value);
+            this.checkboxValue = value;
+            this._setState();
+            this.onChange(this.checkboxValue);
+        }
+    }
+
+    /**
+     * @hidden gets form status and convert to stateType
+     */
+    get stateStatus(): Status | stateType {
+        const formStatus = this.status;
+        if (formStatus === 'error' || formStatus === 'warning') {
+            this._stateStatus = formStatus;
+        } else {
+            this._stateStatus = 'default';
+        }
+        return this._stateStatus;
+    }
+
+    /** Stores current checkbox value. */
+    public checkboxValue: any;
+
+    /** Stores current checkbox state. */
+    private checkboxState: 'checked' | 'unchecked' | 'indeterminate' | 'force-checked';
+
+    /** Values returned by control. */
+    public values: FdCheckboxValues = { trueValue: true, falseValue: false, thirdStateValue: null };
```

Comment:
I am not sure about these values ..

Reply:
implementation is changed now. now only checked value is expected from user.